server/battleships/use_cases/implementation/src/main/java/eu/optas/use_cases/implementation/ShootInteractor.java

```diff
@@ -0,0 +1,90 @@
+package eu.optas.use_cases.implementation;
+
+import eu.optas.domain.Board;
+import eu.optas.domain.Game;
+import eu.optas.domain.Ship;
+import eu.optas.domain.ShotResult;
+import eu.optas.gateway.api.GameGateway;
+import eu.optas.use_cases.api.BoundaryShotResult;
+import eu.optas.use_cases.api.ShootUC;
+import eu.optas.utils.GameState;
+
+import java.util.List;
+
+import static java.util.Objects.nonNull;
+
+public class ShootInteractor implements ShootUC {
+    private final GameGateway gameGateway;
+    private final ShotResultD2BConverter shotResultD2BConverter;
+
+    public ShootInteractor(GameGateway gameGateway, ShotResultD2BConverter shotResultD2BConverter) {
+        this.gameGateway = gameGateway;
+        this.shotResultD2BConverter = shotResultD2BConverter;
+    }
+
+    @Override
+    public BoundaryShotResult shoot(String gameId, int x, int y) throws Exception {
+        Game game = gameGateway.getGame(gameId);
+        if (nonNull(game)) {
+            return shotResultD2BConverter.convert(checkShot(game, x, y));
+        } else throw new Exception("Game doesn't exist!");
```

Comment:
nemėtyk checked exceptionų (tie kur kai kvieti metodą prašo būtinai suhandlint juos), rinkis RuntimeException kaip base klasę ir geriausia, kad susikurtum pats savo exception su pavadinimu kuris nusako situaciją (Uncle Bob irgi šitą turėjo minėt) ir tada teste nereikės handlint to checked exceptiono